packages/docusaurus-utils/src/index.js

```diff
@@ -39,14 +40,34 @@ function encodePath(userpath) {
     .join('/');
 }
 
-function fileToComponentName(file) {
-  const ext = extRE.exec(file)[1];
-  let str = file.replace(extRE, '');
-  str = str.replace(/([A-Z])/g, ' $1');
-  str = str.replace(/^[\W_]+|[\W_]+$/g, '').toLowerCase();
-  str = str.charAt(0).toUpperCase() + str.slice(1);
-  str = str.replace(/[\W_]+(\w|$)/g, (_, ch) => ch.toUpperCase());
-  return ext ? ext.toUpperCase() + str : str;
+/**
+ * Given an input string, convert to kebab-case and append a hash. Avoid str collision
+ * @param {string} str input string
+ * @returns {string}
+ */
+function docuHash(str) {
+  const shortHash = createHash('md5')
+    .update(str)
+    .digest('hex')
+    .substr(0, 3);
+  return `${_.kebabCase(str)}-${shortHash}`;
```

Comment:
The index pages gives `-666.json`. I wonder if we can do it like `genComponentName` below where you return index for an empty path/string.